HealthcareApp/src/Healthcare.Application/Abstractions/IAppointmentRepository.cs

```diff
@@ -0,0 +1,14 @@
+using Domain.Entities;
+
+namespace Application.Abstractions;
+
+public interface IAppointmentRepository
+{
+    Task RequestAppointmentAsync(Appointment appointment);
+    Task<List<Appointment>> GetAllAppointmentsAsync();
+    Task<List<Appointment>> GetRequestedAppointments(string doctorId);
+    Task<Appointment?> GetByIdAsync(int appointmentId);
+    Task SubmitAppointmentAsync(int appointmentId);
+    Task RejectAppointmentAsync(int appointmentId);
+    Task CompleteAppointmentAsync(int appointmentId);
```

Comment:
You seem to pass too much business context to your repositories.
Everything would be ok if you added methods like Submin, Reject, Complete etc to your service level. But repository is about data access - CRUD operations.
Please read more about repository pattern and its use cases